Source/SwiftLintBuiltInRules/Rules/Idiomatic/UnusedEnumeratedRule.swift

```diff
@@ -18,43 +18,211 @@ struct UnusedEnumeratedRule: Rule {
+                onePosition: firstTokenIsUnderscore ? nil : secondElement.positionAfterSkippingLeadingTrivia
+            )
+        }
+
+        override func visit(_ node: FunctionCallExprSyntax) -> SyntaxVisitorContinueKind {
+            guard node.isEnumerated,
+                  let parent = node.parent,
+                  parent.as(MemberAccessExprSyntax.self)?.declName.baseName.text != "filter",
+                  let trailingClosure = parent.parent?.as(FunctionCallExprSyntax.self)?.trailingClosure
+            else {
+                return .visitChildren
+            }
+
+            if let parameterClause = trailingClosure.signature?.parameterClause {
+                guard let parameterClause = parameterClause.as(ClosureShorthandParameterListSyntax.self),
+                      parameterClause.count == 2,
+                      let firstElement = parameterClause.first,
+                      let secondElement = parameterClause.last
+                else {
+                    return .visitChildren
+                }
+
+                let firstTokenIsUnderscore = firstElement.isUnderscore
+                let lastTokenIsUnderscore = secondElement.isUnderscore
+                guard firstTokenIsUnderscore || lastTokenIsUnderscore else {
+                    return .visitChildren
+                }
+
+                addViolation(
+                    zeroPosition: firstTokenIsUnderscore ? firstElement.positionAfterSkippingLeadingTrivia : nil,
+                    onePosition: firstTokenIsUnderscore ? nil : secondElement.positionAfterSkippingLeadingTrivia
+                )
             } else {
-                position = secondElement.positionAfterSkippingLeadingTrivia
+                nextClosureId = trailingClosure.id
+                lastEnumeratedPosition = node.enumeratedPosition
+            }
+
+            return .visitChildren
+        }
+
+        override func visit(_ node: ClosureExprSyntax) -> SyntaxVisitorContinueKind {
+            if let nextClosureId, nextClosureId == node.id, let lastEnumeratedPosition {
+                closures.push(Closure(id: nextClosureId, enumeratedPosition: lastEnumeratedPosition))
+                self.nextClosureId = nil
+                self.lastEnumeratedPosition = nil
+            } else {
+                closures.push(Closure(id: node.id))
+            }
+            return .visitChildren
+        }
+
+        override func visitPost(_ node: ClosureExprSyntax) {
+            guard let closure = closures.peek(), node.id == closure.id else {
+                return
+            }
+            defer { closures.pop() }
+
+            guard (closure.zeroPosition != nil) != (closure.onePosition != nil) else {
+                return
+            }
+
+            addViolation(
+                zeroPosition: closure.zeroPosition,
+                onePosition: closure.onePosition,
+                enumeratedPosition: closure.enumeratedPosition
+            )
```

Comment:
The case that the closure IDs don't match seem impossible. Simplify: 
```suggestion
            if let closure = closures.pop(), (closure.zeroPosition != nil) != (closure.onePosition != nil) {
                addViolation(
                    zeroPosition: closure.zeroPosition,
                    onePosition: closure.onePosition,
                    enumeratedPosition: closure.enumeratedPosition
                )
            }
```